### PR TITLE
ENH: Camviewer Cams

### LINKED ISSUE #117: Camera Objects from Camviewer Config
## Expected Behavior

Check the camviewer config for the hutch and load the relevant area detector objects

## Context

Requested by Silke and Jason

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

Load camviewer.cfg file and create detector objects for the hutch

## Motivation and Context

closes #117 

## How Has This Been Tested?

Added tests

## Where Has This Been Documented?

Small note in docs and docstrings

### CODE REVIEW

**Comment by ZLLentz** [comment]
This is stalled pending the issues raised in https://github.com/pcdshub/pcdsdevices/pull/225 getting resolved.

**Comment by ZLLentz** [comment]
This is done now @teddyrendahl! Reads the camviewer config file and makes all the cameras. It makes them in parallel using a threadpool because cameras that fail from timeouts will fail in series otherwise. You may consider a future upgrade to happi that does the same thing, but with our current no-get-on-init rules this only is really needed with the areadetector stuff. Is camviewer config reading a thing that should make it into happi? Or is this fine as a semi-permanent home for the code?

**Comment by teddyrendahl** [comment]
Hmm. A very good question. Honestly, in my head, this flow of information should be reversed. The camviewer configuration should automatically be generated by `happi`. This isn't the world we live in today so I think adding this PR to `hutch-python` is fine. For L2S-I, when we have a controls system that intimately needs `happi` to function we can re-route the flow of information. By that time however, the camviewer will look very different so I don't think it is worth splitting hairs about now.

**Comment by ZLLentz** [comment]
Yes. This has no way of knowing what cameras are already in the environment, since there is no signal registry.

**Comment by ZLLentz** [comment]
Note to self: revisit and make sure we at least don't create two of the same device ourself! This happens in the XPP file because there is a slow-rate and fast-rate entry for a camera in the cfg.

**Comment by ZLLentz** [comment]
@teddyrendahl this is ready for final review

File: hutch_python/load_conf.py [hunk 1]
```diff
@@ -222,6 +228,18 @@ def load_conf(conf, hutch_dir=None):
...
+        with safe_load('elog'):
+            # Use the fact if we we used the default_platform or not to decide
+            # whether we are in a specialty station or not
+            if default_platform:
+                logger.debug("Using primary experiment ELog")
+                kwargs = dict()
+            else:
+                logger.info("Configuring ELog to post to secondary experiment")
+                kwargs = {'station': '1'}
+            cache(elog=HutchELog.from_conf(hutch.upper(), **kwargs))
```
Comment (teddyrendahl) [inline hutch_python/load_conf.py]: What is this?

Comment (ZLLentz) [reply hutch_python/load_conf.py]: - moved elog earlier in the creation process for organization sake
- skip elog if no hutch e.g. a raw `hutch-python` call

Comment (teddyrendahl) [reply hutch_python/load_conf.py]: Cool.

**Comment by teddyrendahl** [comment]
@ZLLentz I just realized that if you used the `happi.loader` we would catch that you are loading the same object twice. You however have to then go `cfg` file -> `happi.Container` -> `from_container(...)`.

Worth the complication?

**Comment by ZLLentz** [comment]
It's worth the complication so long as we can multithread the happi loads. This cam load process is painful single-threaded because the area detector cameras need to connect to pvs one at a time, complete with timeouts.

**Pull request merged**